simple-todo-app/todoDebot.sol

```diff
@@ -31,37 +42,140 @@ interface ITodo {
 
 
 contract TodoDebot is Debot {
-    address m_address; // TODO contract address
-    Stat m_stat;       // Statistics of incompleted and completed tasks
-    uint32 m_taskId;   // Task id for update. I didn't find a way to make this var local
+    TvmCell m_todoCode; // TODO contract code
+    address m_address;  // TODO contract address
+    Stat m_stat;        // Statistics of incompleted and completed tasks
+    uint32 m_taskId;    // Task id for update. I didn't find a way to make this var local
+    uint256 m_masterPubKey; // User pubkey
+    address m_msigAddress;  // User wallet address
+
+    uint32 INITIAL_BALANCE =  200000000;  // Initial TODO contract balance
+
+
+    function setTodoCode(TvmCell code) public {
+        require(msg.pubkey() == tvm.pubkey(), 101);
+        tvm.accept();
+        m_todoCode = code;
+    }
+
 
     function onError(uint32 sdkError, uint32 exitCode) public {
         Terminal.print(0, format("Operation failed. sdkError {}, exitCode {}", sdkError, exitCode));
         _menu();
     }
 
-    function onSuccess() public {
-        Terminal.print(0, "Transaction succeeded.");
+    function onSuccess() public view {
         _getStat(tvm.functionId(setStat));
     }
 
     function start() public override {
-        AddressInput.get(tvm.functionId(enterTodoAddr),"Enter your TODO contract address");
+        Terminal.input(tvm.functionId(savePublicKey),"Please enter your public key",false);
+    }
+
+    function savePublicKey(string value) public {
+        uint res;
+        bool status;
+        (res, status) = stoi("0x"+value);
+        if (status) {
+            m_masterPubKey = res;
+
+            Terminal.print(0, "Checking if you already have TODO list ...");
+            TvmCell deployState = tvm.insertPubkey(m_todoCode, m_masterPubKey);
+            m_address = address.makeAddrStd(0, tvm.hash(deployState));
+            Terminal.print(0, format( "Info: your TODO contract address is {}", m_address));
+            Sdk.getAccountType(tvm.functionId(checkStatus), m_address);
+
+        } else {
+            Terminal.input(tvm.functionId(savePublicKey),"Wrong public key. Try again!\nPlease enter your public key",false);
+        }
+    }
+
+
+    function checkStatus(int8 acc_type) public {
+        if (acc_type == 1) { // acc is active and  contract is already deployed 
+            _getStat(tvm.functionId(setStat));
+
+        } else if (acc_type == -1)  { // acc is inactive
+            Terminal.print(0, "You don't have TODO list yet, so new contract with an initial balance of 0.2 will be deployed");
```

Comment:
You don't have a TODO list yet, so a new contract with an initial balance of 0.2 tokens will be deployed